src/types/nbodysystem.rs

```diff
@@ -98,4 +127,19 @@ impl NBodySystem {
                 },
             )
     }
+
+    pub fn set_limit(&mut self, limit: u16) {
+        self.m_limit = limit;
+    }
+
+    fn check_limit_reached(&mut self) {
+        if self.m_step <= self.m_limit {
+            self.m_step += 1;
+        } else {
+            self.m_step = 0;
+            let _ = self.signal_sender.send(NBodySignal::LimitReached {
+                particles: self.m_particles.clone(),
+            });
+        }
+    }
```

Comment:
- self.m_step == self.m_limit will result in +1, and reset will occur at limit+1
- If limit == 0, there will be constant spam.

```suggestion
    fn check_limit_reached(&mut self) {
        self.m_step += 1;
        if self.m_step >= self.m_limit {
            self.m_step = 0;
            let _ = self.signal_sender.send(NBodySignal::LimitReached {
                particles: self.m_particles.clone(),
            });
        }
    }
```

Reply:
Done